### PR TITLE
workflows: get DTS_VER from tag not dts-distro.conf

### PR DESCRIPTION
**Pull request by tym2k1**
Test will fail if conflicting versions are present in the push tag and `meta-dts/meta-dts-distro/conf/distro/dts-distro.conf`.

Needs testing before merging

### CODE REVIEW

**Comment by m-iwanicki** [comment]
Won't those changes need to be also added to `develop.yaml`?

**Comment by tym2k1** [comment]
You're right. Added the same check to `.develop.yml`  (but with stripping of the `-*` part) and adjusted the syntax of steps to fetch the DTS_VER from GH Actions environmental variable.

**Comment by tym2k1** [comment]
[Here's a successfull run of `develop` workflow](https://github.com/Dasharo/meta-dts/actions/runs/9992535037/job/27618098880)

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -66,12 +66,26 @@ jobs:
...
+      - name: Validate DTS_VER
+        if: ${{ success() }}
+        shell: bash
+        run: |
+          # Retrieve DTS_VER from previous step
+          DTS_VER_EXPECTED="${{ env.DTS_VER }}"
+          DTS_VER_ACTUAL=$(cat meta-dts/meta-dts-distro/conf/distro/dts-distro.conf | grep DISTRO_VERSION | tr -d "\" [A-Z]_=")
+
+          if [ "${DTS_VER_EXPECTED}" != "${DTS_VER_ACTUAL}" ]; then
+            echo "Expected DTS_VER (${DTS_VER_EXPECTED}) does not match DTS_VER (${DTS_VER_ACTUAL}) in `meta-dts/meta-dts-distro/conf/distro/dts-distro.conf`"
```
Comment (m-iwanicki) [inline .github/workflows/ci.yml:82]:
```suggestion
            echo "Tag (${DTS_VER_EXPECTED}) does not match version in `meta-dts/meta-dts-distro/conf/distro/dts-distro.conf` (${DTS_VER_ACTUAL})"
```

File: .github/workflows/develop.yml [hunk 2]
```diff
@@ -65,12 +65,26 @@ jobs:
...
+      - name: Validate DTS_VER
+        if: ${{ success() }}
+        shell: bash
+        run: |
+          # Retrieve DTS_VER from previous step
+          DTS_VER_EXPECTED="${{ env.DTS_VER }}"
+          DTS_VER_ACTUAL=$(cat meta-dts/meta-dts-distro/conf/distro/dts-distro.conf | grep DISTRO_VERSION | tr -d "\" [A-Z]_=")
+
+          if [ "${DTS_VER_EXPECTED}" != "${DTS_VER_ACTUAL}" ]; then
+            echo "Expected DTS_VER (${DTS_VER_EXPECTED}) does not match DTS_VER (${DTS_VER_ACTUAL}) in `meta-dts/meta-dts-distro/conf/distro/dts-distro.conf`"
```
Comment (m-iwanicki) [inline .github/workflows/develop.yml:81]:
```suggestion
            echo "Tag (${DTS_VER_EXPECTED}) does not match version in `meta-dts/meta-dts-distro/conf/distro/dts-distro.conf` (${DTS_VER_ACTUAL})"
```

**Review by m-iwanicki — CHANGES_REQUESTED** [review]
Error should probably explain why this error happens in a way that doesn't require someone to look into code.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (tym2k1) [reply .github/workflows/ci.yml]: [Adressed](https://github.com/Dasharo/meta-dts/blob/d70e66972cc8514cb6056fcd96130ed18e1db3d7/.github/workflows/ci.yml#L82)

File: .github/workflows/develop.yml (hunk 2 above)
Comment (tym2k1) [reply .github/workflows/develop.yml]: [Adressed](https://github.com/Dasharo/meta-dts/blob/d70e66972cc8514cb6056fcd96130ed18e1db3d7/.github/workflows/develop.yml#L81)

**Comment by tym2k1** [comment]
> Looks good to me

Please resolve all change requests then

**Comment by m-iwanicki** [comment]
@tym2k1 done

**Pull request merged**